### DISCUSSION TITLE
Which property should I target to control the angle of model rotation with keyboard arrows?

### DISCUSSION THREAD

**Discussion by rebrob** [google/model-viewer#4870] [General]
Hello,

Currently, model rotation can be controlled with keyboard arrows (thanks for that!). The default angle of rotation seems to be 22.5 degrees. I would like to change the angle of rotation to 90 degrees (so that we can provide model descriptions for A11y, without having to provide a unique description in 10 degree increments). Is there a particular property that can be targeted to control that angle of rotation?

**Comment by elalish** [comment]
I'm not quite sure which angle you're referring to, but pretty much everything is configurable. Have you tried https://modelviewer.dev/docs/index.html#entrydocs-stagingandcameras-attributes-cameraOrbit and https://modelviewer.dev/examples/loading/#useA11y ?

**Reply by rebrob** [reply]
I have targeted most of those camera control props for other purposes. The specific use case I am referring to is when the user is navigating the model with their keyboard. If they tab to the model, they can use the keyboard arrows to rotate the model. The default rotation for each click seems to be about 22.5 degrees (takes 16 clicks to rotate the model 360 degrees). I would like to override that 22.5 degree setting to 90 degrees. 

I have been looking through the code base and thus far have only found code that controls auto rotation. It seems like that angle of rotation (cameraOrbit )is much smaller than 20 degrees. Do you have any idea which code class controls keyboard rotation (cameraOrbit) via keyPress?

**Reply by elalish** [reply]
Ah, I see now. I'm afraid that increment is hard-coded at the moment. PRs welcome - perhaps we should multiply it by https://modelviewer.dev/docs/#entrydocs-stagingandcameras-attributes-orbitSensitivity, since we already do something similar for zoom.

**Reply by rebrob** [reply]
Ok. Great idea! Thanks for your help.
